tests/upgrades/test_containers_removal.py

```diff
@@ -0,0 +1,205 @@
+"""Tests verifying that Containers support is properly removed
+
+:Requirement: Upgraded Satellite
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: ContainerManagement
+
+:TestType: Functional
+
+:CaseImportance: Medium
+
+:Upstream: No
+"""
+from fauxfactory import gen_string
+
+from nailgun import entities
+from robottelo import ssh
+from robottelo.test import APITestCase, settings
+from robottelo.vm import VirtualMachine
+from upgrade_tests import post_upgrade, pre_upgrade
+from upgrade_tests.helpers.scenarios import (
+    create_dict,
+    get_entity_data,
+)
+
+
+class Scenario_containers_support_removal(APITestCase):
+    """"Verify containers run before upgrade are still running
+    after upgrade, despite foreman_docker removal.
+
+    Test Steps:
+
+            1. Before Satellite upgrade.
+            2. Create docker host
+            3. Create and run container from dockerhub
+            4. Create and run container from external registry
+            5. Upgrade Satellite
+            6. Verify that upgrade procedure removed foreman_docker support
+            7. Verify container from dockerhub is still running
+            8. Verify container from external registry is still running
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(Scenario_containers_support_removal, cls).setUpClass()
+
+    def _vm_cleanup(self, hostname=None):
+        """ Cleanup the VM from provisioning server
```

Comment:
I have one question to everyone, as per my knowledge we do not run post upgrade if the pre-upgrade test for the respective class is failed. In this case, the VM resource will be the lingering state. Should we also add validation such that kill the VM resource if those are created by failed pre-upgrade test? 

I'm not sure about our CI job take care of killing all lingering VM resources.

Reply:
I think you are right. Another test that uses `VirtualMachine` directly, `test_remoteexecution.py`, basically wraps entire pre_upgrade test in `try except` block and destroys vm on any exception: https://github.com/SatelliteQE/robottelo/blob/master/tests/upgrades/test_remoteexecution.py#L91

Let me do similar thing here.

Reply:
Done in 75ce0caebdbece643c26e1d3109a362954d4dd71 (commit diff is misleading, as I had to indent most of the code; but it's really 4 new lines)

Reply:
NO Problem.